prefetch.bs

```diff
@@ -415,3 +424,40 @@ The <dfn>list of sufficiently strict speculative navigation referrer policies</d
       Otherwise it is uncredentialed, and redirects which would return to the original partition (thus ought to have credentials) will cause the prefetch to fail.
     </div>
 </div>
+
+<h2 id="sec-purpose-header">The `Sec-Purpose` HTTP request header</h2>
+
+The `` `Sec-Purpose` `` HTTP request header specifies that the request serves one or more purposes other than requesting the resource for immediate use by the user.
+
+The header field is an [[RFC8941]] Structured Header whose value must be a a <a spec="RFC8941">List</a>. Its ABNF is:
+
+```
+Sec-Purpose = sf-list
+```
+
+It may contain a <a spec="RFC8941">Item</a> member which is the <a spec="RFC8941">Token</a> "`prefetch`". If so, this indicates the request's purpose is to download a resource it is anticipated will be fetched shortly.
+
+<div class="issue">TODO: Are there normative implications of this that should be specified here?</div>
+
+The following parameters is defined for the "`prefetch`" member:
+
+* A parameter whose key is "`anonymous-client-ip`".
+
+  If present with a value other than boolean false, this parameter indicates that the prefetch request is being made using an anonymous client IP. Consequently, servers should not rely on it matching, or sharing a geographic location or network operator with, the client's IP address from which a non-prefetch request would have been made.
+
+  If a suitable response is not possible, for example because the resource depends on the client's geographic location, there is no other means of determining the location (e.g., the <a href="https://www.ietf.org/archive/id/draft-geohash-hint-00.html">Geohash client hint</a>), and no location-agnostic response is available, then the server should respond with an appropriate HTTP status code and response headers which mark the response as not suitable for caching.
+
+  <div class="note">
+    A future specification might define assign more specific meaning to non-boolean values. For now, they are treated the same as true. Implementations are advised not to emit such values.
+  </div>
+
+<h2 id="iana-considerations">IANA Considerations</h2>
+
+This document registers the "`Sec-Purpose`" header in the <a href="https://www.iana.org/assignments/http-fields/http-fields.xhtml">Hypertext Transfer Protocol (HTTP) Field Name Registry</a>:
```

Comment:
We could maybe skip straight to https://github.com/protocol-registries/http-fields these days. Recently HTML and Fetch have removed all their header registration appendices in favor of that.

Reply:
So the modern thing to do is to not include the registration appendix at all, and merely to file a GitHub issue there at an appropriate time? (When is an appropriate time for "wide review"? When we've resolved a couple of the TODOs added in this PR? Immediately upon this landing?)

Reply:
I'm not really sure what the protocol is for new headers like this. I think it hasn't really been tried; we've only converted old established headers. So I'd say it's up to you to attempt registration whenever you feel it's worthwhile.

Reply:
Alright, filed #118 to do this in the future and removed the IANA considerations section from here for now.